pkg/registry/controller/rest.go

```diff
@@ -47,14 +48,26 @@ func (rcStrategy) NamespaceScoped() bool {
 func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
+	controller.Spec.SequenceNumber = 0
+	controller.Spec.SequenceNumber = 0
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (rcStrategy) PrepareForUpdate(obj, old runtime.Object) {
-	// TODO: once RC has a status sub-resource we can enable this.
-	//newController := obj.(*api.ReplicationController)
-	//oldController := old.(*api.ReplicationController)
-	//newController.Status = oldController.Status
+	newController := obj.(*api.ReplicationController)
+	oldController := old.(*api.ReplicationController)
+
+	// Any changes to the spec increment the sequence number, any changes to the
+	// status reflect the sequence number of the corresponding spec
+	if !reflect.DeepEqual(oldController.Spec, newController.Spec) {
+		newController.Spec.SequenceNumber += 1
+	}
+	if !reflect.DeepEqual(oldController.Status, newController.Status) {
+		// newController is the controller from the client, oldController is the current
+		// controller. It doesn't matter which one we use here, because the update will
+		// fail if they're not the same anyway.
+		newController.Status.SequenceNumber = oldController.Spec.SequenceNumber
```

Comment:
We trust the controller.

Eventually we'd want to make status writable only by the controller.